design/replace-CLI-with-SDK.md

```diff
@@ -0,0 +1,220 @@
+
+<!-- toc -->
+- [Motivation](#motivation)
+- [Scope](#scope)
+- [High Level Design](#high-level-design)
+  - [Kopia SDK wrappers](#kopia-sdk-wrappers)
+    - [Storage pkg](#storage-pkg)
+    - [Repository pkg](#repository-pkg)
+  - [Repository server controller changes](#repository-server-controller-changes)
+	- [Kopia CLI Approach](#kopia-cli-approach)
+	- [Kopia SDK Approach](#kopia-sdk-approach)
+<!-- /toc -->
+
+
+# Motivation
+
+Kanister uses a kubernetes custom controller makes use of kopia as a primary backup and restore tool. 
+The detailed design of the customer controller can be found [here](https://github.com/kanisterio/kanister/blob/master/design/kanister-kopia-integration.md)
+
+The custom controller called as repository server controller currently uses kopia CLI to perform the kopia operations. All the
+operations are executed inside a pod using the `kubectl exec` function. `kubectl exec` can be flaky for long running operations
+and gives less control over command that is being executed. 
+
+The goal over here is to build a kopia server programatically and reduce the dependency on kopia CLI in turn reducing the usage of 
```

Comment:
Technically we are not implementing Kopia server, right? We are implementing a proxy service that serves APIs through which clients can setup a Kopia server? Is that right? We should reframe this better.

Reply:
we wont be running any other service in the pod apart from kopia service. The pod would be started by executing a pre-built binary. The binary would execute all the operations required to start the kopia server.